### PR TITLE
Move CI testing to Ubuntu

### PR DESCRIPTION
**Pull request by Hinton**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [x] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Updates the CI testing workflows to use a Ubuntu OS for running the tests. This seems to result in a  ~10min speed up of our tests, going from 30min to 20min. I also disabled the logs for our integration tests since they provide little to no value.

To allow this to build properly on Ubuntu I remove the `Sql` project from the solution in the pipeline prior to executing the tests.

## Before you submit

- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- If making database changes - make sure you also update Entity Framework queries and/or migrations
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -64,20 +61,23 @@ jobs:
...
         shell: pwsh
 
       - name: Test OSS solution
-        run: dotnet test ./test --configuration Debug --no-build --logger "trx;LogFileName=oss-test-results.trx" || true
+        run: dotnet test ./test --configuration Debug --no-build --logger "trx;LogFileName=oss-test-results.trx"
         shell: pwsh
 
       - name: Test Bitwarden solution
-        run: dotnet test ./bitwarden_license/test --configuration Debug --no-build --logger "trx;LogFileName=bw-test-results.trx" || true
+        run: dotnet test ./bitwarden_license/test --configuration Debug --no-build --logger "trx;LogFileName=bw-test-results.trx"
```
Comment (Hinton) [inline .github/workflows/build.yml:80]: @withinfocus Any reason we did `|| true` here initially?

Comment (withinfocus) [reply .github/workflows/build.yml:80]: You want to let the test run failure still continue the process here to catch said test failures. The reporter will also emit a failure to the Action.

Example: https://www.octopus.com/blog/githubactions-running-unit-tests#:~:text=The%20test%20command%20returns%20a%20non%2Dzero%20exit%20code%20if%20any%20tests%20fail%2C%20but%20you%20defer%20responsibility%20for%20responding%20to%20failed%20tests%20to%20the%20test%20processor.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -64,20 +61,23 @@ jobs:
         run: dotnet restore --locked-mode
         shell: pwsh
 
+      - name: Remove SQL proj
+        run: dotnet sln bitwarden-server.sln remove src/Sql/Sql.sqlproj
+
       - name: Build OSS solution
-        run: msbuild bitwarden-server.sln /p:Configuration=Debug /p:DefineConstants="OSS" /verbosity:minimal
+        run: dotnet build bitwarden-server.sln -p:Configuration=Debug -p:DefineConstants="OSS" --verbosity minimal
```
Comment (withinfocus) [inline .github/workflows/build.yml:68]: ⚠️ Moving away from MSBuild can sometimes have unintended consequences with outputs, or at least it used to. Worth being sure about.

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -64,20 +61,23 @@ jobs:
         run: dotnet restore --locked-mode
         shell: pwsh
 
+      - name: Remove SQL proj
+        run: dotnet sln bitwarden-server.sln remove src/Sql/Sql.sqlproj
```
Comment (withinfocus) [inline .github/workflows/build.yml:65]: ❓ What's the failure? I thought this just warned.

This project type really isn't compatible anymore and Microsoft doesn't seem to want to add its support across the ecosystem outside Windows and VS proper, so I'd support its removal as a project and we manage the files manually. Yes, there are some benefits to its use with compile-like feedback of SQL issues, but if you're not using VS proper (which almost no one here is I hear) then you don't get them.

**Review by withinfocus — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml (hunk 1 above)
Comment (Hinton) [reply .github/workflows/build.yml:80]: That's what the `if always()` is for, per https://github.com/dorny/test-reporter#recommended-setup-for-public-repositories right?

Comment (withinfocus) [reply .github/workflows/build.yml:77]: What I believe I saw is this stoped the process, but I see you trying it out with a failed test -- open to the change.

Comment (Hinton) [reply .github/workflows/build.yml:80]: @withinfocus https://github.com/bitwarden/server/runs/10932209501 seems to work.

File: .github/workflows/build.yml (hunk 3 above)
Comment (Hinton) [reply .github/workflows/build.yml:65]: Seems to be needed.

File: .github/workflows/build.yml (hunk 1 above)
Comment (withinfocus) [reply .github/workflows/build.yml:80]: Good catch!

**Review by withinfocus — APPROVED** [review]

**Review by joseph-flinn — APPROVED** [review]

**Pull request merged**